src/views/Viewer.vue

```diff
@@ -407,6 +424,27 @@ export default {
 				}
 			}
 		},
+
+		isFullscreenMode(mode) {
+			if (!this.isImage) {
+				return
+			}
+
+			// Use 100% of screen height to display images and make the header semitransparent
+			//
+			// NOTE: This fragment depends on internal structure of the NcModal component.
+			//
+			const modalHeader = document.getElementsByClassName('modal-header').item(0)
+			const modalContainer = document.getElementsByClassName('modal-container').item(0)
+			if (mode) {
+				modalHeader.classList.add('modal-header--semitransparent')
+				modalContainer.classList.add('modal-container--fullscreen')
+			} else {
+				modalHeader.classList.remove('modal-header--semitransparent')
+				modalContainer.classList.remove('modal-container--fullscreen')
+			}
```

Comment:
This is the only point that I'm not fond off.
Why not put the class in the Viewer container straight away and use `:deep()` straight away?
That would allow us depending on the internal structure of the NcModal component ? :)

Reply:
Thanks for your suggestion! :+1: I did not even think to do it this way. Implemented in https://github.com/nextcloud/viewer/pull/1588/commits/1d6a18e7fab767fcaf7ef588ae3fbc01a55ce644. Much cleaner indeed.

Although I am still struggling with scoped SCSS, deep selectors, etc. :wink:

Reply:
Yeah, it can be a bit weird to understand it at first :see_no_evil: